_posts/2025-05-22-debezium-as-part-of-your-ai-solution.adoc

```diff
@@ -0,0 +1,438 @@
+
+Debezium Server provides an interface between the PostgreSQL store and the Milvus sink.
+Because embeddings are not provided, we can use the https://github.com/debezium/debezium-examples/blob/main/ai-rag/config-server/application.properties#L16[Debezium FieldToEmbedding SMT] to calculate them and store them a field named `vector`.
+
+Our application uses LangChain4j Milvus store module internally so we will keep the default collection structure https://github.com/langchain4j/langchain4j/blob/main/langchain4j-milvus/src/main/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStore.java[expected by it]
+
+* `id` - `VarChar`
+* `metadata` - `JSON`
+* `text` -`VarChar`
+* `vector` - `FloatVector`
+
+There is a known https://github.com/langchain4j/langchain4j/issues/2834[issue] in the module that causes an application failure when the store is empty.
+We are going to store one dummy document in the database (and thus in Milvus) to work around it.
+
+== AI Application
+
+The AI application is the application that interacts with LLM and extends the prompt with data loaded from Milvus stored base on the data and prompt embeddings similarity.
+
+The demo application is command-line driven, written using the https://quarkus.io/[Quarkus framework].
+For AI features it uses Quarkus https://docs.quarkiverse.io/quarkus-langchain4j/dev/index.html[LangChain4j extension], specifically https://docs.quarkiverse.io/quarkus-langchain4j/dev/ollama.html[Ollama LLM] and https://docs.quarkiverse.io/quarkus-langchain4j/dev/milvus-store.html[Milvus Store] for RAG.
+
+The AI service is registered via the https://github.com/debezium/debezium-examples/blob/main/ai-rag/client/src/main/java/io/debezium/examples/airag/Chat.java[Chat] class.
+
+[source,java]
+----
+package io.debezium.examples.airag;
+
+import dev.langchain4j.service.SystemMessage;
+import dev.langchain4j.service.UserMessage;
+import io.quarkiverse.langchain4j.RegisterAiService;
+
+@RegisterAiService(retrievalAugmentor = MilvusRetrieavalAugmentor.class)
+public interface Chat {
+
+    @SystemMessage("You are an expert that provides short summaries.")
+    String chat(@UserMessage String message);
+}
+----
+
+The module itself is configured in https://github.com/debezium/debezium-examples/blob/main/ai-rag/client/src/main/resources/application.properties[application.properties].
+
+----
+quarkus.langchain4j.ollama.base-url=http://localhost:11434
+quarkus.langchain4j.ollama.chat-model.model-id=granite3.1-dense:2b
+quarkus.langchain4j.ollama.embedding-model.model-id=granite3.1-dense:2b
+quarkus.langchain4j.ollama.timeout=120s
+----
+
+RAG is enabled using the `retrievalAugmentor` parameter on the`@RegisterAiService` annotation and is implemented in the https://github.com/debezium/debezium-examples/blob/main/ai-rag/client/src/main/java/io/debezium/examples/airag/MilvusRetrieavalAugmentor.java[MilvusRetrieavalAugmentor] bean.
```

Comment:
`@RegisterAiService` is not rendered correctly as code, but not sure what's wrong here, probably something antora specific as plain asciidoc renders it correctly.

Reply:
Fixed